ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Combinator/Ledger/Query.hs

```diff
@@ -309,14 +309,19 @@ answerQueryAnytime HardForkLedgerConfig{..} =
 -------------------------------------------------------------------------------}
 
 data QueryHardFork xs result where
```

Comment:
@DavidEichmann A key part of that comment: if you're changing the queries, we'll have to bump `NodeToClientVersion`. @EncodePanda has more experience with that than I do, though we're both pretty new to it.

Reply:
Is there some test that would catch this stuff? E.g. something that looks at the query types / versions.
I actually don't see any versioning code for QueryHardFork :thinking: @EncodePanda, perhaps you know what's going on here?

Reply:
(Pawel and I were preoccupied today with Alonzo stuff. I'm spending a brief time now trying to give you at least some feedback in the meantime.)

We have golden tests to check if existing serializations change (edit: maybe not for HFC queries?!), but nothing to check for eg new queries, AFAIK. Maybe we could somehow reformulate the generators in `ouroboros-consensus-cardano-test/src/Test/Consensus/Cardano/Generators.hs` so that the compiler will warn us if we're missing a query constructor (ie just put a dummy `case` expression in there, with explanatory comment).

Here's an example of where the queries are version-gated; hopefully that gives you enough to chase the existing patterns; let us know if not.

https://github.com/input-output-hk/ouroboros-network/blob/0fbc52ea24f5c111f5814d87a2dd9f10190be45d/ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Combinator/Serialisation/SerialiseNodeToClient.hs#L186

Reply:
The latest commit adds such warnings and bumps the version number. Question! I plan on adding more hard fork queries after this PR. Will I need a new protocol version for each PR?!? Or do we only need to bump the version between releases? How does the release cycle even work?

Reply:
> Or do we only need to bump the version between releases? How does the release cycle even work?

Yeah, when merging a PR that touches this, we check to see if the latest version has been included in a release yet. I've only been involved for that for one such case. So my best advice is to consult others, esp Duncan, when the time comes.